conformance/lang/expressions/unary-expression/unary-complement-expression.balt

```diff
@@ -0,0 +1,483 @@
+    int res = ~a3;
+    io:println(res); // @output -1001
+
+    io:println(~int:SIGNED32_MIN_VALUE);  // @output 2147483647
+    io:println(~int:SIGNED32_MAX_VALUE);  // @output -2147483648
+}
+
+Test-Case: error
+Description: Test unary complement expression assignments resulting in an error.
+Labels: byte, int:Unsigned8, int:Unsigned16, int:Unsigned32, int:Signed8, int:Signed16,langlib, int:Signed32,
+module-type-defn, unary-complement
+
+type Ints -1|-2;
+
+public function main(byte x1, int:Unsigned8 x2, int:Unsigned16 x3, int:Unsigned32 x4, int:Signed8 x5,
+                    int:Signed16 x6, int:Signed32 x7) {
+    byte res1 = ~x1; // @error static type of complement expression with operand of byte is int
+
+    int:Unsigned8 res2 = ~x2; // @error static type of complement expression with operand of Unsigned8 is int
+
+    int:Unsigned16 res3 = ~x3; // @error static type of complement expression with operand of Unsigned16 is int
+
+    int:Unsigned32 res4 = ~x4; // @error static type of complement expression with operand of Unsigned32 is int
+
+    int:Signed8 res5 = ~x5; // @error static type of complement expression with operand of Signed8 is int
+
+    int:Signed16 res6 = ~x6; // @error static type of complement expression with operand of Signed16 is int
+
+    int:Signed32 res7 = ~x7; // @error static type of complement expression with operand of Signed32 is int
+
+    Ints x8 = ~1; // @error static type of complement expression with operand of int subtype is int
+}
+
+Test-Case: error
+Description: Test the static type of operand belonging to non-int type used as a constant resulting in an error.
+Fail-Issue: ballerina-lang#32628
+Labels: const, float, unary-complement
+
+const int CONST1 = ~false; // @error unary complement not supported for boolean
+const int CONST2 = ~~false; // @error unary complement not supported for boolean
+
+const int CONST3 = ~"false"; // @error unary complement not supported for string
+const int CONST4 = ~~"false"; // @error unary complement not supported for string
+
+float floatVal = 1.5;
+const int CONST5 = ~floatVal; // @error unary complement not supported for float
+
+public function main() {
+}
+
+Test-Case: error
+Description: Test the static type of operand belonging to non-int type resulting in an error.
+Labels: float, tuple-type, unary-complement
+
+public function main() {
+    int a1 = ~true; // @error unary complement not supported for boolean
+
+    [int, int] intTuple = [1, 2];
+    int a3 = ~intTuple; // @error unary complement not supported for [int, int]
+
+    float floatVal = 1.5;
+    int a4 = ~floatVal; // @error unary complement not supported for float
+
+    int a5 = ~~false; // @error unary complement not supported for boolean
+}
+
+Test-Case: output
+Description: Test the static type of operand belonging to int optional type.
+Fail-Issue: ballerina-lang#31701
+Labels: optional-type, unary-complement
+
+public function main() {
+    int? result = ~getIntOrNil();
```

Comment:
Shall we add operand being nil case also. Like this
```
int? a = ();
a = ~a;
```

Reply:
And also shall we add similar test cases for other integer sub types

Reply:
Added in 1a98eb5fcbb2ae216fc85f8a4692fd098e9a45a9